### DISCUSSION TITLE
How to add support for a new language that is in endangered language category

### DISCUSSION THREAD

**Discussion by khera-shanu** [openai/whisper#1162] [Ideas]
I want to do it for the Tangkhul language - [Wiki Page](https://en.wikipedia.org/wiki/Tangkhul_language)

It doesn't have its own writing system/alphabet and uses the Latin alphabet, also there is not a lot of written content except 2-3 books and a few dictionaries, and one book explaining its grammar as someone's Ph.D. Research.

What is the minimum I will need for training whisper to identify a language like that, I am new to AI but not programming, also passionate to solve this problem just for the sake of solving it.

I don't mind if someone points me in the right direction not looking for spoon-feeding, Thanks for reading.

P.S. There are more than 200k people who use this language and you can assume 10% of them to have access to an internet connection and smart phone, open to ideas for crowd sourcing for data collection.

**Comment by jongwook** [comment]
To leverage Whisper, you'd require a few hours of recordings with labels, split into segments shorter than 30 seconds.

- #364 
- #759 
- [The HuggingFace blog](https://huggingface.co/blog/fine-tune-whisper)

I think the use of latin alphabets would make fine-tuning marginally easier, but still there will be a lot of research efforts needed for the transcription to work reliably. [Somewhat related effort was done for Hindi recently](https://officechai.com/stories/indian-govt-releases-version-of-openais-whisper-model-which-turns-hindi-speech-into-text/).
